src/wl_window.c

```diff
@@ -610,8 +610,12 @@ static void toplevel_configure(void *data, struct xdg_toplevel *toplevel,
   (void)toplevel;
 
   struct imv_window *window = data;
-  window->width = width;
-  window->height = height;
+
+  if (width > 0 && height > 0)
```

Comment:
Can you merge lines 614 and 615 to match the style of the other `if` statements. Otherwise this looks good. Thanks for the PR.